lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -362,54 +371,70 @@ def test_clear_tools_test86(tools_configuration, groups, remotes, tools):
     out, err, exitcode = pytest.helpers.capture(command)
     assert b"" == out
 
-    empty_group_err_msg = "Blank group name specified; skipping."
-    empty_remote_err_msg = "Blank remote host name specified; skipping."
-    empty_tool_err_msg = "Blank tool name specified; skipping."
+    empty_group_err_msg = (
+        "Error: Invalid value for '-g' / '--group' / '--groups': "
+        "Blank group name specified; terminating.\n"
+    )
+    empty_remote_err_msg = (
+        "Error: Invalid value for '-r' / '--remote' / '--remotes': "
+        "Blank remote name specified; terminating.\n"
+    )
+    empty_tool_err_msg = (
+        "Error: Invalid value for '-n' / '--name' / '--names': "
+        "Blank tool name specified; terminating.\n"
+    )
 
     if not groups:
         groups = "default"
 
-    empty_group_seq = 0
-    empty_remote_seq = 0
-    empty_tool_seq = 0
-    for group in groups.split(","):
-        invalid_group_err_msg = f'pbench-clear-tools: invalid --group option "{group}"'
-        if not group:
-            empty_group_seq += 1
-        elif group not in tools_tree.keys():
-            assert invalid_group_err_msg.encode() in err
+    groups_list = groups.split(",")
+    remotes_list = remotes.split(",")
+    tools_list = tools.split(",")
+
+    # Check for the blank group name
+    if [group for group in groups_list if not group]:
```

Comment:
As Dave pointed out, this would be nicer as
```
    if not all(groups_list):
```
Analogously for lines 399 and 403, and, I don't think you need to guard against the list being empty, because `not all([])` is `False`, so you'll take the `else` branch, anyway.